client/common/cmdline.c

```diff
@@ -589,7 +589,14 @@ static char** freerdp_command_line_parse_comma_separated_values_ex(const char* n
 
 			if (p)
 			{
-				p[0] = name;
+				p[0] = strdup(name);
```

Comment:
why the duplication here?

Reply:
Because name is declared cons char*, but the function returns char** p.
name comes from calls to freerdp_command_line_parse_comma_separated_values_offset with either literal arguments, or arg->Name.  So it's better to create a mutable string in all cases.

The alternative would be to return a const char**  but this result is used in a dozen or more places, where const would have to be added.

Granted, most of the strings in those argv arrays are never mutated, only the passwords are (AFAIK), but if we want the help of the compiler validating that, using const or not, we need to do this. (unfortunately, we cannot specify that some slots of the array are mutable and some are not; we'd have to use a struct for that).

Reply:
@informatimago that is a memory leak, sorry.